py3status/modules/xrandr_toggle.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+"""
+Toggles a monitor on or off from the status bar using xrandr.
+
+Configuration parameters:
+    cache_timeout: how often we refresh usage in seconds (default: 10s)
+    format: see placeholders below
+    format_disconnected: custom text or symbol for disconnected status
+    format_off: custom text or symbol for off status
+    format_on: custom text or symbol for on status
+    number: number of the monitor we want to query
+    position: where the monitor will be toggled (ex: left-of HDMI-1)
+    source: name of the source we want to query (ex: HDMI-2)
+
+Format of status string placeholders:
+    {screen} screen source name
+    {status} on/off status
+
+Example config:
+
+```
+xrandr_toggle {
+    # we could use the monitor number from xrandr output
+    # name = "1"
+    # or the monitor id directly
+    source = "HDMI-2"
+    position = "left-of DVI-I-1"
+    format = "[{screen}:TV32\"]"
+}
+```
+
+@author mrt-prodz
+"""
+from subprocess import call, check_output
+from re import search
+from time import time
+
+
+class Py3status:
+    # available configuration parameters
+    cache_timeout = 10
+    format = '{source}:{status}'
+    format_disconnected = 'disconnected'
+    format_on = 'on'
+    format_off = 'off'
+    number = '1'
+    position = ''
+    source = ''
+
+    def __init__(self):
+        self.mon = None
+        self.status = self._xrandr()
+
+    # return error occurs
+    def _error_response(self, color, text):
+        response = {
+            'cached_until': time() + self.cache_timeout,
+            'full_text': 'xrandr_toggle: ' + text,
+            'color': color
+        }
+        return response
+
+    # query xrandr for monitors status
+    def _xrandr(self):
+            return check_output(['xrandr']).split('\n')
+
+    # return connected monitor using xrandr and number for index
+    def get_status(self, i3s_output_list, i3s_config):
+        try:
+            # monitor number (index) must be bigger than 0
+            number = int(self.number) - 1
+            if number < 0:
+                return self._error_response(i3s_config['color_bad'],
+                                            'invalid number')
+            monitors = []
+            # width x height + left + top
+            regex = r'\d{1,}x\d{1,}\+\d{1,}\+\d{1,}'
+            # for each line find (dis)connected monitors
+            for line in self.status:
+                if 'connected' in line:
+                    details = line.split(' ')
+                    if len(details) > 2:
```

Comment:
This is happening even in bash if I call `xrandr` too often it will slow down and the mouse will freeze for a couple ms. I was following the xrandr module at first and noticed that issue.

Do you think it takes some time for `xrandr` to query all output and result in a delay to refresh workspaces? I could try in another environment to see if there is a similar issue.